sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.ark.container.command;
+
+import java.util.Set;
+
+import com.alipay.sofa.ark.container.service.ArkServiceContainerHolder;
+import com.alipay.sofa.ark.spi.model.Plugin;
+import com.alipay.sofa.ark.spi.service.plugin.PluginManagerService;
+
+/**
+ * telnet plugin info provider
+ *
+ * @author joe
+ * @version 2018.07.20 11:45
+ */
+public class PluginCommandProvider extends AbstractCommandProvider {
+    private static String        COMMAND_PRE = "plugin";
+    private PluginManagerService service     = ArkServiceContainerHolder.getContainer().getService(
+                                                 PluginManagerService.class);
+
+    @Override
+    protected String getCommandPre() {
+        return COMMAND_PRE;
+    }
+
+    @Override
+    protected String handleCommandLine(String[] commands) {
+        String command = commands[1];
+        StringBuilder sb = new StringBuilder();
+
+        switch (command) {
+            case "list":
+                if (commands.length == 2) {
+                    sb.append("plugins :\n").append(getPrint(service.getAllPluginNames(), "\t"));
+                    break;
+                }
+            case "info":
+                if (commands.length == 3) {
+                    Plugin plugin = service.getPluginByName(commands[2]);
+                    if (plugin == null) {
+                        sb.append(String.format("plugin %s not found", commands[2]));
+                    } else {
+                        sb.append(getPrint(plugin));
+                    }
+                    break;
+                }
+            default:
+                sb.append("command error\n");
```

Comment:
`command error` should be evaluated in validate phase.

Reply:
This line will not run now, but if do not add this default there will be warning.